napari/_qt/qt_main_window.py

```diff
@@ -270,10 +271,17 @@ def _save_current_window_settings(self):
         if settings.application.save_window_state:
             settings.application.window_state = window_state
 
-    def close(self, quit_app=False):
+    def close(self, quit_app=False, confirm_need=False):
```

Comment:
perhaps in this PR you'd like to tackle the difference in behavior depending on how a close is triggered that was noticed before?  I think this logic might want to go in `closeEvent`? (which gets called regardless of whether the window is closed via button, shortcut, etc...)

Reply:
I do not know how to distinguish the call `close()` by file menu and shortcut from this one called from script/tests. `confirm_need` is used for this. I do not see a way to do this form `closeEvent` as it may be called from Qt event loop. 

Or do you have an idea how to do this?

Reply:
@tlambert03 does this need to be addressed before merge?

Reply:
No, but would be good to not let it slip away.  There's an open issue for it